addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -1423,6 +1423,20 @@ export class OdooEditor extends EventTarget {
             this.dispatchEvent(new Event('historyRevert'));
         }
     }
+    /**
+     * Ensure that a callback is called without triggering a rollback.
+     *
+     * If a rollback was set before the callback, do not reset it.
+     */
+    withoutRollback(callback) {
+        if (this._toRollback) {
+            callback();
```

Comment:
Wait... we're going to rollback this at some point anyway, right ? Since `this._toRollback` is truthy. So what's the point of executing the callback ? I feel like what this is doing is protecting against a reset of `this._toRollback` from a previous call but then I think this form would be clearer:
```js
const priorRollback = this._toRollback;
callback();
if (this._toRollback && !priorRollback) {
    this.observerFlush();
    this._toRollback = false;
}
```
That being said, my original question in fact still stands: What's the point of executing the callback since it's going to be rollbacked anyway ? I guess we could have another call to `withoutRollback` inside the callback, in which case we'd have to execute the function so I think it is in fact more correct to call the `callback` function even though we might think that it is going to be rollbacked.

Reply:
Your version would not work as for `this._toRollback` to be set to a truthy value, we need to call `this.observerFlush()` after the callback.

So that would be:
```js
const priorRollback = this._toRollback;
callback();
this.observerFlush();
if (!priorRollback) {
    this._toRollback = false;
}
```

For your point about the execution of the callback if `_toRollback` is truthy: if the callback have other side effect than altering the dom, we would prevent those side effect. I'm not sure this is what you'd expect from the `withoutRollback` method.

Reply:
Change done.